src/pages/home/report/ReportActionsView.js

```diff
@@ -91,14 +89,13 @@ class ReportActionsView extends React.Component {
         this.keyboardEvent = Keyboard.addListener('keyboardDidShow', this.scrollToListBottom);
         this.recordMaxAction();
         fetchActions(this.props.reportID);
+        this.setUpUnreadActionIndicator();
+        Timing.end(CONST.TIMING.SWITCH_REPORT, CONST.TIMING.COLD);
     }
 
     shouldComponentUpdate(nextProps, nextState) {
-        if (nextProps.reportID !== this.props.reportID) {
-            return true;
-        }
-
         if (!_.isEqual(nextProps.reportActions, this.props.reportActions)) {
+            this.updateSortedReportActions(nextProps.reportActions);
```

Comment:
More than a year ago that this was added now, but wondering if there was any particular reason this side-effect was put in `shouldComponentUpdate` instead of `componentDidUpdate`?

Reply:
Is it because you were hoping to avoid having to evaluate `_.isEqual(nextProps.reportAction, this.props.reportActions)` in both `shouldComponentUpdate and `componentDidUpdate`?

Reply:
Not exactly... `componentDidUpdate()` happens _after_ the `render()` so (as things are now) if we sorted the actions there then we'd also need to trigger an re-render somehow.

Using `shouldComponentUpdate()` with a class property is a way to re-sort actions _before_ the `render()` method is called.

I guess a more current approach could be to recalculate the options in the render with a dependency on `props.reportActions`, but my hook fu is pretty weak tbh 😂 

![DZ-97vzW4AAbcZj](https://user-images.githubusercontent.com/32969087/193165377-09c5801f-1e7b-4976-8c11-6165583f273f.jpeg)

Reply:
> More than a year ago that this was added now, but wondering if there was any particular reason this side-effect was put in `shouldComponentUpdate` instead of `componentDidUpdate`?

Marc already answered, but yes, to capture `sortedActions` before rendering and without triggering a re-render 

> I guess a more current approach could be to recalculate the options in the render with a dependency on `props.reportActions`

If your talking about hooks it would be as simple as 

```js
const sortedActions = useMemo(() => util.sortActions(props.reportActions), [props.reportActions])
```

But then the component would need to be converted from a class to a function based component (to use hooks)

For a class component the only way to sort actions, (only) when the prop changed, is to use `shouldComponentUpdate`

---

Another alternative is to subscribe the view to `reportId` actions, and sort them in the `connection` configuration, so the component always receives sorted actions (they'd still be sorted only if they change)

Reply:
Yep, that's sort of what I had in mind. But, I think `_.isEqual()` is doing an optimized deep comparison whereas `useMemo()` would only run if the `reportActions` failed a simple object comparison? (so even better)

>Another alternative is to subscribe the view to reportId actions, and sort them in the connection configuration, so the component always receives sorted actions (they'd still be sorted only if they change)

That's pretty interesting. If we are doing a `keyChanged()` then we know the subscriber is getting new data and it's an opportunity to run some logic without having to do the deep compare.

That said, I think we are maybe always creating new objects when merging or setting changes so when calling `keysChanged()` the simple compare in `useMemo()` might be all that we'd need. When keyChanged is called the object reference for something like `props.reportActions` would be pointing to a new object.

Reply:
This conversation here is slightly related -> https://github.com/Expensify/react-native-onyx/pull/185

Reply:
> Another alternative is to subscribe the view to reportId actions, and sort them in the connection configuration, so the component always receives sorted actions (they'd still be sorted only if they change)

This is pretty interesting, but I think overall the hook-based approach might be simpler/more accessible to more people in the long term?

Anyways, thanks for the thorough answers here!